test/e2e/framework/pod/wait.go

```diff
@@ -868,3 +868,18 @@ func WaitForContainerTerminated(ctx context.Context, c clientset.Interface, name
 		return false, nil
 	})
 }
+
+// WaitForContainerRestartedNTimes waits for the given Pod container to have restarted N times
+func WaitForContainerRestartedNTimes(ctx context.Context, c clientset.Interface, namespace, podName, containerName string, timeout time.Duration, target int) error {
+	conditionDesc := fmt.Sprintf("container %s restarted at least %d times", containerName, target)
+	return WaitForPodCondition(ctx, c, namespace, podName, conditionDesc, timeout, func(pod *v1.Pod) (bool, error) {
+		for _, statuses := range [][]v1.ContainerStatus{pod.Status.ContainerStatuses, pod.Status.InitContainerStatuses, pod.Status.EphemeralContainerStatuses} {
+			for _, cs := range statuses {
```

Comment:
nit: use `podutil.GetStatus(...)`

Reply:
I'm not sure exactly which util you're talking about. I did find [podutils.maxContainerRestarts](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/util/podutils/podutils.go#L194) which I could use instead, given I know I only have one container in this pod.

Reply:
I guess maxContainerRestarts works... I was referring to [`GetContainerStatus`](https://github.com/kubernetes/kubernetes/blob/252e9cbb236af91bb60c3ae7fda08bb31e1f4d9b/pkg/api/v1/pod/util.go#L272) (sorry I had the name wrong)

Reply:
ahhh ok. I like that better

Reply:
fixed in https://github.com/kubernetes/kubernetes/pull/128559/commits/8e7b2af712a79db423b7bfdf2a7bf35f75542c13